test/integration/unit_tests/test_utility.py

```diff
@@ -40,5 +40,17 @@ def test_get_execution_duration_from_global_result_json_missing_suffix():
     utility.get_execution_duration_from_global_result_json(global_result_json)
 
 
+def test_count_log_lines_with_substring_counts_only_lines_with_substring():
```

Comment:
Thank you for adding the coverage, should we also add a test case with zero lines and one which has lines, but none of them matches?

Reply:
Done.